src/Core/Content/Mail/Message/SendMailHandler.php

```diff
@@ -0,0 +1,60 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Content\Mail\Message;
+
+use League\Flysystem\FilesystemOperator;
+use Psr\Log\LoggerInterface;
+use Shopware\Core\Framework\Log\Package;
+use Symfony\Component\Mailer\Exception\TransportExceptionInterface;
+use Symfony\Component\Mailer\Transport\TransportInterface;
+use Symfony\Component\Messenger\Attribute\AsMessageHandler;
+use Symfony\Component\Mime\Email;
+
+#[AsMessageHandler]
+#[Package('services-settings')]
+class SendMailHandler
+{
+    /**
+     * @internal
+     */
+    public function __construct(
+        private readonly TransportInterface $transport,
+        private readonly FilesystemOperator $filesystem,
+        private readonly LoggerInterface $logger
+    ) {
+    }
+
+    /**
+     * @throws TransportExceptionInterface
+     */
+    public function __invoke(SendMailMessage $message): void
+    {
+        $mailDataPath = $message->getMailDataPath();
+
+        if (!$this->filesystem->fileExists($mailDataPath)) {
+            $this->logger->error('An error occurred while sending an email. The mail data file does not exist.');
+
+            return;
+        }
+
+        $mailData = $this->filesystem->read($mailDataPath);
+
+        $mail = unserialize($mailData);
+
+        if (!($mail instanceof Email)) {
+            $this->filesystem->delete($mailDataPath);
+
+            $this->logger->error('An error occurred while sending an email. The mail data is invalid.');
+
+            return;
+        }
+
+        $this->transport->send($mail);
```

Comment:
The envelope is gone?

Reply:
The MailSender has a internal construct and is only used by the MailService which never passes the envelope object. So I guess we don't need it anymore?